crypto/keyring/keyring.go

```diff
@@ -823,6 +848,79 @@ func (ks keystore) writeMultisigKey(name string, pub types.PubKey) (Info, error)
 	return info, nil
 }
 
+func (ks keystore) checkMigrate() error {
+	var version uint32 = 0
+	item, err := ks.db.Get(VERSION_KEY)
+	if err != nil {
+		if err != keyring.ErrKeyNotFound {
+			return err
+		}
+		// key not found, all good: assume version = 0
+	} else {
+		if len(item.Data) != 4 {
+			return sdkerrors.ErrInvalidVersion.Wrapf(
+				"Can't migrate the keyring - the stored version is malformed: [%v]: %v",
+				item.Description, string(item.Data))
+		}
+		version = binary.LittleEndian.Uint32(item.Data)
+	}
+	return ks.migrate(version, item)
+}
+
+func (ks keystore) migrate(version uint32, i keyring.Item) error {
+	if version == CURRENT_VERSION {
+		return nil
+	}
+	if version > CURRENT_VERSION {
+		return sdkerrors.ErrInvalidVersion.Wrapf(
+			"Can't migrate the keyring - wrong keyring version: [%v]: %v, expected version to be max %d",
+			i.Description, string(i.Data), CURRENT_VERSION)
+	}
+	keys, err := ks.db.Keys()
+	if err != nil {
+		return err
+	}
+
+	for _, key := range keys {
+		if !strings.HasSuffix(key, infoSuffix) {
+			continue
+		}
+		item, err := ks.db.Get(key)
+		if err != nil {
+			return err
+		}
+
+		if len(item.Data) == 0 {
+			return sdkerrors.Wrap(sdkerrors.ErrKeyNotFound, key)
+		}
+	
+		info, err := protoUnmarshalInfo(item.Data, ks.cdc)
+		if err != nil {
+			return err
+		}
+
+		var versionBytes = make([]byte, 4)
+		binary.LittleEndian.PutUint32(versionBytes, CURRENT_VERSION)
+		ks.db.Set(keyring.Item{
+			Key:         info.GetKey().String(),
+			Data:        versionBytes,
+			Description: "SDK kerying version",
```

Comment:
As noted above, we store `Info` here, not a version bytes. So: `Description` is not needed, `Data` must be changed.